compiler-rt/lib/scudo/standalone/combined.h

```diff
@@ -1594,14 +1598,19 @@ class Allocator {
     RB->StackDepotSize = StackDepotSize;
     RB->RawStackDepotMap = DepotMap;
 
-    atomic_store(&RingBufferAddress, reinterpret_cast<uptr>(RB),
-                 memory_order_release);
+    // If multiple threads try to initialize at the same time, let one thread
+    // win and throw away the work done in the other threads. Since this
+    // path is only meant for debugging, a race that results in work being
+    // discarded should not matter.
+    uptr EmptyPtr = 0;
+    if (!atomic_compare_exchange_strong(&RingBufferAddress, &EmptyPtr,
+                                        reinterpret_cast<uptr>(RB),
+                                        memory_order_acquire)) {
+      unmapRingBuffer(RB);
+    }
```

Comment:
I would suggest doing the initialization under a lock so that if there's any issue during buffer allocation, we won't see several failures with same error (I know it's unlikely according to the use case but still want to avoid any confusion if possible)

We may also want to ensure the enable()/disable() take/release the init-lock respectively

Reply:
I added a lock, but just in the init function. I think that should be sufficient to prevent multiple threads trying to init at the same time.